File: verify_app.py
from playwright.sync_api import sync_playwright
import os

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    
    page.on("console", lambda msg: print(f"PAGE LOG: {msg.text}"))
    page.on("pageerror", lambda err: print(f"PAGE ERROR: {err}"))
    
    url = "file://" + os.path.abspath("index.html")
    print(f"Navigating to {url}")
    page.goto(url)
    
    page.wait_for_timeout(2000)
    page.screenshot(path="screenshot.png")
    browser.close()

File: verify_app.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  page.on('console', msg => console.log('PAGE LOG:', msg.text()));
  page.on('pageerror', err => console.log('PAGE ERROR:', err.message));

  console.log('Navigating to local app...');
  await page.goto('file://' + process.cwd() + '/index.html');
  
  // Wait for Firebase to maybe fail or something
  await page.waitForTimeout(2000);
  
  await page.screenshot({ path: 'screenshot.png' });
  await browser.close();
})();
